### PR TITLE
Update README with rationale

### PR DESCRIPTION
**Pull request by chris-crone**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,175 @@
...
+While they are generally unrestricted key-value pairs, [some guidance is given
+for keys by the OCI](https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys).
+
+Formalising some common annotations across various artifact types will make them
+useful for registries to use to better understand the artifacts.
+
+### The future
+
+There is a clear trend towards more types of artifacts being stored in container
+registries. While it's too early to predict exactly what will be stored in
```
Comment (silvin-lubecki) [inline README.md]: nit: s/it's/it is

File: README.md [hunk 2]
```diff
@@ -2,16 +2,175 @@
...
+
+A container image is presented by a registry as a
+[manifest](https://github.com/opencontainers/image-spec/blob/master/manifest.md).
+Each manifest is platform specific which means that in order to use an image on
+multiple platforms, one needs to fetch the correct manifest for that platform.
+
+Initially this was solved by indicating the platform as part of the tag, e.g.:
+`myimage:tag-<platform>`. This is undesirable for base images used on multiple
+platforms as it requires platform specific code. As such a manifest list was
+added where multiple manfiests could be presented behind the same code.
```
Comment (silvin-lubecki) [inline README.md]: typo: manifests

**Comment by silvin-lubecki** [comment]
Can you rebase on top of master? I fixed the metalinter.

File: README.md [hunk 3]
```diff
@@ -214,6 +375,28 @@ The following is an example of an OCI image index sent to the registry.
...
+    "size": 578
+  },
+  "layers": []
+}
+```
+
+Subsequent manifests in the manifest list are standard OCI images.
+
+This example proposes two OCI specification and registry changes:
+1. It proposes the addition of an `org.opencontainers.artifactType` annotation to be included in the OCI specification.
```
Comment (SteveLasker) [inline README.md]: This is where we're centering on `manifest.config.mediaType` as the declaration of it's type

File: README.md [hunk 4]
```diff
@@ -172,32 +331,33 @@ The following is an example of an OCI image index sent to the registry.
 
 ```json
 {
-  "schemaVersion": 1,
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
```
Comment (simonferquel) [inline README.md]: mediaType is not there inside OCI manifests

File: README.md [hunk 5]
```diff
@@ -214,6 +375,28 @@ The following is an example of an OCI image index sent to the registry.
 }
 ```
 
+The first manifest in the manifest list references the CNAB configuration. An
+example of this follows:
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
```
Comment (simonferquel) [inline README.md]: same here

**Comment by chris-crone** [comment]
Fixed @simonferquel, PTAL

**Comment by radu-matei** [comment]
Thank you for this document!

**Pull request merged**